src/docfx/lib/log/Error.cs

```diff
@@ -47,12 +47,16 @@ public Error(ErrorLevel level, string code, string message, FilePath? file = nul
             Name = name;
         }
 
-        public Error WithCustomError(CustomError customError)
+        public Error WithCustomRule(CustomRule customRule, bool? isCanonicalVersion = null)
```

Comment:
consider promote `IsCanonicalVersion` into a property (similar to `Name`) so you don't need to use a tuple [here](https://github.com/dotnet/docfx/pull/6035/files#diff-eb444f860ada42127995887c1b00a00aR519)

Reply:
👍